flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -347,43 +351,88 @@ public static String getTaskManagerShellCommand(
 			boolean hasKrb5,
 			Class<?> mainClass) {
```

Comment:
There were some subtle differences in the memory setup (is that really intended?)), log and config directories and files which is why I did not generalise the setup.
It may be done, however... question is whether it is easier to grasp afterwards. Or let's worry about this when there are three setups of this kind ;)